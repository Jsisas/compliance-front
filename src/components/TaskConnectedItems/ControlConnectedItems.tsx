import * as React from 'react';
import {Col, Row, Typography} from "antd";
import {Link} from "react-router-dom";
import {Regulation} from "../../redux/Regulation/RegulationSlice";
import {Task} from "../../redux/Task/TaskSlice";
import {Requirement} from "../../redux/Requirement/RequirementSlice";
import {Control} from "../../redux/Control/ControlSlice";
import {lowerCameltoUpperCamel, shortenStringLength} from "../../util/StringUtil";
import styles from './contorlConnectedItems.module.scss';

const {Text} = Typography;

interface TaskConnectedItemsProps {
    task: Task;
}

export function TaskConnectedItems(props: TaskConnectedItemsProps) {

    return (
        <>
            <Col xs={24} className={styles.connectedItem}>
                <Row gutter={[16, 16]}>
                    <Col xs={24}>
                        <Text style={{fontWeight: 600}}>Connected items</Text>
                    </Col>
                </Row>
                <Row gutter={[16, 0]}>
                    <Col xs={24}>
                        <Text type={'secondary'}>Control</Text>
                    </Col>
                </Row>
                <Row gutter={[16, 16]}>
                    <Col xs={24}>
                        <Link key={props.task.control.id} to={`/controls/${props.task.control.id}`}>
                            {shortenStringLength(props.task.control.title, 50)}
                        </Link>
                    </Col>
                </Row>
            </Col>
        </>
    );
};