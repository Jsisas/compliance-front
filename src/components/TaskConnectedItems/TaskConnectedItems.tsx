import * as React from 'react';
import {Col, Row, Typography} from "antd";
import {Link} from "react-router-dom";
import {Task} from "../../redux/Task/TaskSlice";
import styles from './taskConnectedItems.module.scss';
import StringUtil from "../../util/StringUtil";

const {Text} = Typography;

interface TaskConnectedItemsProps {
    task: Task;
}

export function TaskConnectedItems(props: TaskConnectedItemsProps) {

    return (
        <>
            <Col xs={24} className={styles.connectedItem}>
                <Row gutter={[16, 16]}>
                    <Col xs={24}>
                        <Text style={{fontWeight: 600}}>Connected items</Text>
                    </Col>
                </Row>
                <Row gutter={[16, 0]}>
                    <Col xs={24}>
                        <Text type={'secondary'}>Control</Text>
                    </Col>
                </Row>
                <Row gutter={[16, 16]}>
                    <Col xs={24}>
                        <Link key={props.task.control.id} to={`/controls/${props.task.control.id}`}>
                            {StringUtil.shortenStringLength(props.task.control.title, 50)}
                        </Link>
                    </Col>
                </Row>
            </Col>
        </>
    );
};
