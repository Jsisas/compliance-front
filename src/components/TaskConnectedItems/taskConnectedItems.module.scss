.connectedItem{
  background: #FFFFFF;
  border: 1px solid #DCDDDD;
  box-sizing: border-box;
  border-radius: 4px;
}
