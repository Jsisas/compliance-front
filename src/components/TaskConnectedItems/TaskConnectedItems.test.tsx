import { deepRender } from '../../util/test/TestUtil';
import React from 'react';
import { TaskConnectedItems } from './TaskConnectedItems';
import { testTasks } from '../../util/test/data/DataUtil';

it('it matches a snapshot', () => {
	const [history, element] = deepRender(<TaskConnectedItems task={testTasks[0]} />);
	expect(element).toMatchSnapshot();
});
