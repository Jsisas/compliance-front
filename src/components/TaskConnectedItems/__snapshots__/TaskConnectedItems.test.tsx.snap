// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches a snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-col connectedItem ant-col-xs-24"
      >
        <div
          class="ant-row"
          style="margin: -8px -8px 8px -8px;"
        >
          <div
            class="ant-col ant-col-xs-24"
            style="padding: 8px 8px 8px 8px;"
          >
            <span
              class="ant-typography"
              style="font-weight: 600;"
            >
              Connected items
            </span>
          </div>
        </div>
        <div
          class="ant-row"
          style="margin-left: -8px; margin-right: -8px;"
        >
          <div
            class="ant-col ant-col-xs-24"
            style="padding-left: 8px; padding-right: 8px;"
          >
            <span
              class="ant-typography ant-typography-secondary"
            >
              Control
            </span>
          </div>
        </div>
        <div
          class="ant-row"
          style="margin: -8px -8px 8px -8px;"
        >
          <div
            class="ant-col ant-col-xs-24"
            style="padding: 8px 8px 8px 8px;"
          >
            <a
              href="/controls/495d2960-fd50-4756-a7d9-7a3b8a082552"
            >
              Definition of security roles and responsibilities
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-col connectedItem ant-col-xs-24"
    >
      <div
        class="ant-row"
        style="margin: -8px -8px 8px -8px;"
      >
        <div
          class="ant-col ant-col-xs-24"
          style="padding: 8px 8px 8px 8px;"
        >
          <span
            class="ant-typography"
            style="font-weight: 600;"
          >
            Connected items
          </span>
        </div>
      </div>
      <div
        class="ant-row"
        style="margin-left: -8px; margin-right: -8px;"
      >
        <div
          class="ant-col ant-col-xs-24"
          style="padding-left: 8px; padding-right: 8px;"
        >
          <span
            class="ant-typography ant-typography-secondary"
          >
            Control
          </span>
        </div>
      </div>
      <div
        class="ant-row"
        style="margin: -8px -8px 8px -8px;"
      >
        <div
          class="ant-col ant-col-xs-24"
          style="padding: 8px 8px 8px 8px;"
        >
          <a
            href="/controls/495d2960-fd50-4756-a7d9-7a3b8a082552"
          >
            Definition of security roles and responsibilities
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
